assets/differentialabundance_report.Rmd

```diff
@@ -838,32 +880,67 @@ for (i in 1:nrow(contrasts)){
       }
     }
   }
-  
 }
 ```
 
 <!-- Gene set analysis results -->
 
 ```{r, echo=FALSE, results='asis'}
-possible_gene_set_methods <- c('gsea')
+possible_gene_set_methods <- c('gsea', 'gprofiler2')
 if (any(unlist(params[paste0(possible_gene_set_methods, '_run')]))){
   cat("\n### Gene set analysis\n")
 
   for (gene_set_method in possible_gene_set_methods){
     if (unlist(params[paste0(gene_set_method, '_run')])){
       cat("\n#### ", toupper(gene_set_method) ," {.tabset}\n")
-      
-      for (gmt_file in simpleSplit(params$gsea_gene_sets)) {
-        gmt_name <- basename(tools::file_path_sans_ext(gmt_file))
-        cat("\n##### ", gmt_name ," {.tabset}\n")
-        reference_gsea_tables <- paste0(contrasts$id, ".", gmt_name, '.gsea_report_for_', contrasts$reference, '.tsv')
-        target_gsea_tables <- paste0(contrasts$id, ".", gmt_name, '.gsea_report_for_', contrasts$target, '.tsv')
-        for (i in 1:nrow(contrasts)){
-          cat("\n###### ", contrast_descriptions[i], "\n")
-          target_gsea_results <- read_metadata(target_gsea_tables[i])[,c(-2,-3)]
-          print( htmltools::tagList(datatable(target_gsea_results, caption = paste0("\nTarget (", contrasts$target[i], ")\n"), rownames = FALSE) ))
-          ref_gsea_results <- read_metadata(reference_gsea_tables[i])[,c(-2,-3)]
-          print( htmltools::tagList(datatable(ref_gsea_results, caption = paste0("\nReference (", contrasts$reference[i], ")\n"), rownames = FALSE) ))
+      if (gene_set_method == 'gsea') {
+        for (gmt_file in simpleSplit(params$mods_gene_sets)) {
+          gmt_name <- basename(tools::file_path_sans_ext(gmt_file))
+          cat("\n##### ", gmt_name ," {.tabset}\n")
+
+          reference_gsea_tables <- paste0(contrasts$id, ".", gmt_name, '.gsea_report_for_', contrasts$reference, '.tsv')
+          target_gsea_tables <- paste0(contrasts$id, ".", gmt_name, '.gsea_report_for_', contrasts$target, '.tsv')
+          for (i in 1:nrow(contrasts)){
+            cat("\n###### ", contrast_descriptions[i], "\n")
+            target_gsea_results <- read_metadata(target_gsea_tables[i])[,c(-2,-3)]
+            print( htmltools::tagList(datatable(target_gsea_results, caption = paste0("\nTarget (", contrasts$target[i], ")\n"), rownames = FALSE) ))
+            ref_gsea_results <- read_metadata(reference_gsea_tables[i])[,c(-2,-3)]
+            print( htmltools::tagList(datatable(ref_gsea_results, caption = paste0("\nReference (", contrasts$reference[i], ")\n"), rownames = FALSE) ))
+          }
+        }
+
+      } else if (gene_set_method == 'gprofiler2') {
+        enrichment_files <- grep("gprofiler2", list.files(params$input_dir), value=T, fixed=T)
```

Comment:
Can you be consistent with the GSEA loop please? So loop through the gmt files and contrasts in the same way, and infer what the gprofiler files should be from the contrast names (they should be predictable). Then you can say explicitly when things are missing for a given contrast or gene sets file.

Reply:
The GMT file is optional for gprofiler; it does not behave exactly like the GSEA module. I'll have to think how to still make the loop work

Reply:
So gprofiler has internal databases it would use without gmt files? Fair enough then- but still iterate over contrasts and look for matching files I think.